docfx/doxygen_groups.cc

```diff
@@ -0,0 +1,174 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "docfx/doxygen_groups.h"
+#include "docfx/doxygen2markdown.h"
+#include "docfx/doxygen2yaml.h"
+#include "docfx/doxygen_errors.h"
+#include <yaml-cpp/yaml.h>
+#include <iostream>
+#include <sstream>
+
+namespace docfx {
+namespace {
+
+void AppendReferences(YAML::Emitter& yaml, pugi::xml_node const& node) {
+  for (auto const& child : node) {
+    auto const name = std::string_view{child.name()};
+    if (name == "sectiondef") {
+      AppendReferences(yaml, child);
+      continue;
+    }
+    if (name == "innergroup") {
+      AppendReferences(yaml, child);
+      continue;
+    }
+    if (name == "memberdef") {
+      yaml << std::string{child.attribute("id").as_string()};
+      continue;
+    }
+    if (name == "innerclass") {
+      yaml << std::string{child.attribute("refid").as_string()};
+      continue;
+    }
+  }
+}
+
+}  // namespace
+
+std::string Group2Yaml(Config const& /*config*/, pugi::xml_node const& node) {
+  auto const id = std::string{node.attribute("id").as_string()};
+  YAML::Emitter yaml;
+  yaml << YAML::BeginMap;
+  yaml << YAML::Key << "items" << YAML::Value << YAML::BeginSeq;
+  yaml << YAML::BeginMap;
+  yaml << YAML::Key << "uid" << YAML::Value << id         //
+       << YAML::Key << "name" << YAML::Value << id        //
+       << YAML::Key << "id" << YAML::Value << id          //
+       << YAML::Key << "type" << YAML::Value << "module"  //
+       << YAML::Key << "langs" << YAML::BeginSeq << "cpp" << YAML::EndSeq
+       << YAML::Key << "summary" << YAML::Literal
+       << Group2SummaryMarkdown(node);
+  yaml << YAML::Key << "references" << YAML::BeginSeq;
+  AppendReferences(yaml, node);
+  yaml << YAML::EndSeq;
+  yaml << YAML::EndMap;
+  yaml << YAML::EndSeq << YAML::EndMap;
```

Comment:
Nit: There doesn't seem to be a reason to treat 65/66 differently to 67.

Reply:
Thanks.  PTAL, the change may not be what you had in mind.

Reply:
Even better than what I had in mind.